### ISSUE TITLE
bucket_location not taken into account?

### ISSUE THREAD

**Issue by bramstes** [s3tools/s3cmd#682]
I've only been using it since today but from what I can see the host_bucket and host_base parameters in the configuration file don't take into account the value of the bucket_location.
I've set my bucket_location to eu-west-1.
If I use it like that without further modifications to the host_bucket and host_base parameter values I always get the [Errno 32] from what I see because amazon returns a redirect towards eu-west-1.
When I manually adapt the host_bucket and host_base parameter values to "s3-eu-west-1.amazonaws.com" and "%(bucket)s.s3-eu-west-1.amazonaws.com" it works.
I tried modifying the parameter values to %(bucket)s.s3-%(bucket_location)s.amazonaws.com but that didn't work :-).

Regards,
Bram.

**Comment by bramstes** [comment]
BTW as far as I know I cloned the latest version from git.

**Comment by mdomsch** [comment]
By default those values point at the us-east-1 servers regardless of the
region specified and then get redirected. To avoid the redirect you have to
specify those host values as you have done.

**Comment by mdomsch** [comment]
We do not have code to automatically detect the region endpoint based on
setting --region/bucket_location because of the many S3 clones that would
make such unreasonable. As it happens, lots of people use S3 clones. I am
open to suggestions. There is even an old branch I started to work on it
but abandoned it until a good solution came forward.

**Comment by bramstes** [comment]
Thanks for the reply! To be clear I like what you've made and let us use for free, it works pretty good.
The automatic redirect didn't work in my case (latest macosx, python 2.7) it worked when I tried to put and empty file on Amazon S3, but it continuously failed when I was trying to 'put' a photo (2,5MB) (over HTTPS, didn't try HTTP).
When the small file worked and I saw the redirect in the debug traces I figured I had to put the region in the URL. Then I remembered I had to enter the region during configuration (and I didn't see it used anywhere in the configuration file itself) so I wondered why not add that parameter in host_bucket string and resolve the value when the URL is used. That would indeed only cover the Amazon cases, not the S3 clone cases didn't think about those (I don't see how one can cover those automatically. If one diverts from the Amazon path I would expect manual tweaking.).
Anyway it's a detail, but if the redirect doesn't work one has to know which values to adapt manually.
Thanks again!
